packages/superset-ui-dimension/src/createSVGNode.ts

```diff
@@ -0,0 +1,40 @@
+import { TextStyle } from './types';
+
+export interface CreateSVGNodeInput {
+  className?: string;
+  container?: HTMLElement;
+  style?: TextStyle;
+}
+
+const SVG_NS = 'http://www.w3.org/2000/svg';
+const STYLE_FIELDS: (keyof TextStyle)[] = [
+  'font',
+  'fontWeight',
+  'fontStyle',
+  'fontSize',
+  'fontFamily',
+  'letterSpacing',
+];
+
+export default function createSVGNode(input: CreateSVGNodeInput): SVGSVGElement {
+  const { className, style = {}, container = document.body } = input;
+
+  const textNode = document.createElementNS(SVG_NS, 'text');
+  if (className !== undefined && className !== null) {
+    textNode.setAttribute('class', className);
+  }
+  STYLE_FIELDS.filter(
```

Comment:
You will have to remove className and remove all styles from this text node before continuing. otherwise style will keep accumulating

Reply:
We're only doing this once now, though. Not on every call like the first version of the PR.

Reply:
My point is there is no guarantee that all text in consecutive calls will have the exact same style. and by not clearing them, the later ones will have accumulated style.

Reply:
Sorry, @kristw, I still don't understand.

There's 2 ways of calling `getTextDimension` with this PR:

In the old one, you call it basically with a string and possible a style or class name. In that case:

1. An SVG node is created and added to the passed container.
2. A text node is created and appended to the SVG node.
3. The class is applied to the text node.
4. Any whitelisted (ie, in `STYLE_FIELDS`) style field is applied to the text node.
5. The text node content is set to the string.
6. Its bounding box is computed.
7. The SVG node is removed from the container.

I think we both agree that in this case there's no way to accumulate styles, since the node is created and removed for every call.

In the new one, you call `createSVGNode` yourself, and pass it to `getTextDimension` with a string. When calling `createSVGNode`:

1. An SVG node is created and added to the passed container.
2. A text node is created and appended to the SVG node.
3. The class is applied to the text node.
4. Any whitelisted (ie, in `STYLE_FIELDS`) style field is applied to the text node.
5. The SVG node is returned.

Now you can pass this node together with the string to `getTextDimension`. Note that when the existing node is passed, `className`, `style` and `container` are ignored and have no effect. So now, each call to `getTextDimension` will simply:

1. Set the content of the text node.
2. Compute its bounding box.

When the existing SVG node is reused *we're not* setting the styles nor the class name again, we simply replace the text content and compute the bounding box.